test-lit/lit.cfg.py

```diff
@@ -53,32 +37,32 @@
 config.test_exec_root = os.path.join(config.flang_obj_root, 'test-lit')
 
 config.substitutions.append(('%PATH%', config.environment['PATH']))
-config.substitutions.append(('%shlibext', config.llvm_shlib_ext))
 
 llvm_config.use_default_substitutions()
 
 # excludes: A list of directories to exclude from the testsuite. The 'Inputs'
 # subdirectories contain auxiliary inputs for various tests in their parent
 # directories.
-config.excludes = ['Inputs', 'CMakeLists.txt', 'README.txt', 'LICENSE.txt', 'debuginfo-tests']
+config.excludes = ['CMakeLists.txt', 'README.txt', 'LICENSE.txt']
 
 # test_source_root: The root path where tests are located.
 config.test_source_root = os.path.dirname(__file__)
 
 # test_exec_root: The root path where tests should be run.
-config.test_exec_root = os.path.join(config.flang_obj_root, 'test_lit')
+config.test_exec_root = os.path.join(config.flang_obj_root, 'test-lit')
 
 # Tweak the PATH to include the tools dir.
+llvm_config.with_environment('PATH', config.flang_tools_dir, append_path=True)
 llvm_config.with_environment('PATH', config.llvm_tools_dir, append_path=True)
 
 # For each occurrence of a flang tool name, replace it with the full path to
 # the build directory holding that tool.  We explicitly specify the directories
 # to search to ensure that we get the tools just built and not some random
 # tools that might happen to be in the user's PATH.
-tool_dirs = [config.flang_tools_dir, config.llvm_tools_dir]
-
-tools = ['flang', 'f18', 'opt']
+tool_dirs = [config.llvm_tools_dir, config.flang_tools_dir]
 
+tools = [ToolSubst('%flang', command=FindTool('flang')) ,
```

Comment:
I think you want to add an `unresolved=fatal` to these calls so that they fail if they don't find their tools. I suspect that very few tests will run without flang.